### PR TITLE
Regression testing workflow

### LINKED ISSUE #110: Develop a regression testing workflow
Right now, I'm thinking that we would use test-tiny.xml (or something similar), outputting XML so that we can produce a reference simulation result and then use tools like diff to see if new results match old ones. I think this is feasible for simple simulations on the CPU. Exact reproducibility on the GPU may be more difficult, as it's possible that threads associated with the same neuron in two different simulations may get different RNG-generated noise in two different simulations. But we should be able to take a first step towards this relatively straightforwardly.

### PR DESCRIPTION
**Pull request by jbrown725**
Save known good simulator outputs in a dedicated folder and compare them to outputs generated by our GitHub action workflow. Closes #110

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Unit Tests
+name: Regression Tests
```
Comment (stiber) [inline .github/workflows/tests.yml:1]: Well, the Google tests are unit tests. Maybe "Unit and regression tests"? Then, you can add "regression" to the names of your three tests below.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -21,6 +21,22 @@ jobs:
...
     - name: build
       run: make
-    - name: run tests
-      run: ./tests
+    - name: run unit tests
+      run: ./tests    
+
+    # simulator runs
+    - name: run test-tiny
+      run: ./cgraphitti -c ../configfiles/test-tiny.xml
```
Comment (stiber) [inline .github/workflows/tests.yml:29]: I'm a little concerned about leaving the config files for regression testing in the configfiles directory; various people modify them for various purposes (this is not good practice, but...). Perhaps you could copy them into the `../Testing/RegressionTestComparison` directory?

**Review by stiber — CHANGES_REQUESTED** [review]
See comments above.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (jbrown725) [reply .github/workflows/tests.yml:1]: My thinking was that unit tests and simulator runs (functional tests) are both under the category of regression testing. Both types are run after every change to make sure nothing breaks.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (jbrown725) [reply .github/workflows/tests.yml]: test config files have been copied to their own folder in the Testing directory

File: .github/workflows/tests.yml (hunk 1 above)
Comment (stiber) [reply .github/workflows/tests.yml]: True, but regression testing involves comparison over time, while the unit tests are comparisons with something like "ground truth", such as ensuring that correct files and objects are created.

**Review by stiber — APPROVED** [review]
Good to go!

**Pull request merged**